### PR TITLE
Remove a few awaits when executing cells

### PR DESCRIPTION
**Pull request by DonJayamanne**
Doesn't fix everything, but a simple change to remove some unnecessary awaits.

### CODE REVIEW

File: src/client/datascience/jupyter/kernels/cellExecution.ts [hunk 1]
```diff
@@ -125,7 +135,8 @@ export class CellExecution {
         private readonly kernelConnection: Readonly<KernelConnectionMetadata>,
         disposables: IDisposableRegistry,
         extensionContext: IExtensionContext,
-        private readonly controller: NotebookController
+        private readonly controller: NotebookController,
+        private readonly outuptDisplayIdTracker: CellOutputDisplayIdTracker
```
Comment (IanMatthewHuff) [inline src/client/datascience/jupyter/kernels/cellExecution.ts:139]: output

File: src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import { createDeferred, Deferred } from '../../../common/utils/async';
+
+@injectable()
+export class CellOutputDisplayIdTracker {
+    private displayIdCellOutputMappingPerDocument = new WeakMap<
+        NotebookDocument,
+        Map<string, { output: Deferred<NotebookCellOutput>; cell: NotebookCell }>
+    >();
+    private cellToDisplayIdMapping = new WeakMap<NotebookCell, string>();
+    constructor(@inject(IVSCodeNotebook) notebooks: IVSCodeNotebook) {
```
Comment (rchiodo) [inline src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts:16]: What about when a notebook is closed? Shouldn't the mapping be cleared then?

**Review by rchiodo — APPROVED** [review]

File: src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                    });
+            }
+        });
+    }
+    /**
+     * Keep track of the mapping between display_id and the output.
+     * When we need to update this display, we can resolve the promise & access the output.
+     * The return value is a promise that needs to be resolved with the associated output thats been added to the DOM
+     */
+    public trackOutputByDisplayId(cell: NotebookCell, displayId: string): Deferred<NotebookCellOutput> {
```
Comment (IanMatthewHuff) [inline src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts:38]: I might just need to read this over a second time, but I can't quite tell if it's mapping to the Cell or the Output item here.

Comment (rchiodo) [reply src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts:38]: Both?

Map from document to display id map of cells and a promise
Map from cell to display id

Comment (IanMatthewHuff) [reply src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts:38]: Yeah, I think I got it on the second pass. Wasn't really parsing that it was creating the deferred here in this function.

**Review by IanMatthewHuff — APPROVED** [review]

File: src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts:16]: Its a weakmap, hence no need.

File: src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts (hunk 3 above)
Comment (DonJayamanne) [reply src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts:38]: Happy to refactor to something else

Comment (IanMatthewHuff) [reply src/client/datascience/jupyter/kernels/cellDisplayIdTracker.ts:38]: Nah it's fine just took a bit to parse.

File: src/client/datascience/jupyter/kernels/cellExecution.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/datascience/jupyter/kernels/cellExecution.ts:139]: Will fix in next PR

**Pull request merged**